src/anbox/platform/sdl/platform.cpp

```diff
@@ -341,6 +294,53 @@ void Platform::process_input_event(const SDL_Event &event) {
     touch_->send_events(touch_events);
 }
 
+int Platform::find_touch_slot(int id){
+    for(int i = 0; i < MAX_FINGERS; i++){
+        if(touch_slots[i] == id)return i;
+    }
+    return -1;
+}
+
+void Platform::push_slot(std::vector<input::Event> &touch_events, int slot){
+    if(last_slot != slot){
+        touch_events.push_back({EV_ABS, ABS_MT_SLOT, slot});
+        last_slot = slot;
+    }
+}
+
+void Platform::push_finger_down(int x, int y, int finger_id, std::vector<input::Event> &touch_events){
+    int slot = find_touch_slot(-1);
+    if(slot == -1){
+        DEBUG("no free slot!\n");
```

Comment:
There is no `\n` necessary

Reply:
Fixed. I hope I didn`t miss anything. I am sorry for the delay.

P.